app/post.go

```diff
@@ -883,7 +889,7 @@ func (a *App) GetPostThread(postID string, opts model.GetPostsOptions, userID st
 		}
 	}
 
-	if appErr := a.filterInaccessiblePosts(posts); appErr != nil {
+	if appErr := a.filterInaccessiblePosts(posts, filterPostOptions{assumeSortedCreatedAt: true}); appErr != nil {
```

Comment:
For this one, sorting is dependent on `opts.Direction`, if it has no value then there is no sorting.